bot/exts/evergreen/latex.py

```diff
@@ -0,0 +1,76 @@
+import asyncio
+import re
+from concurrent.futures import ThreadPoolExecutor
+from io import BytesIO
+
+import discord
+import matplotlib.pyplot as plt
+from discord.ext import commands
+
+# configure fonts and colors for matplotlib
+plt.rcParams.update(
+    {
+        "font.size": 16,
+        "mathtext.fontset": "cm",  # Computer Modern font set
+        "mathtext.rm": "serif",
+        "figure.facecolor": "38383F",  # matches Discord's dark mode background color
+        "text.color": "white",
+    }
+)
+
+FORMATTED_CODE_REGEX = re.compile(
+    r"(?P<delim>(?P<block>```)|``?)"  # code delimiter: 1-3 backticks; (?P=block) only matches if it's a block
+    r"(?(block)(?:(?P<lang>[a-z]+)\n)?)"  # if we're in a block, match optional language (only letters plus newline)
+    r"(?:[ \t]*\n)*"  # any blank (empty or tabs/spaces only) lines before the code
+    r"(?P<code>.*?)"  # extract all code inside the markup
+    r"\s*"  # any more whitespace before the end of the code markup
+    r"(?P=delim)",  # match the exact same delimiter from the start again
+    re.DOTALL | re.IGNORECASE,  # "." also matches newlines, case insensitive
+)
+
+
+class Latex(commands.Cog):
+    """Renders latex."""
+
+    @staticmethod
+    def _render(text: str) -> BytesIO:
+        """Return the rendered image if latex compiles without errors, otherwise raise a BadArgument Exception."""
+        fig = plt.figure()
+
+        try:
+            fig.text(0, 1, text, horizontalalignment="left", verticalalignment="top")
+
+            rendered_image = BytesIO()
+            plt.savefig(rendered_image, bbox_inches="tight", dpi=600)
+            rendered_image.seek(0)
+            return rendered_image
+
+        except ValueError as e:
+            raise commands.BadArgument(str(e))
```

Comment:
Which line is actually going to raise this ValueError? Ideally we would only wrap that individual line in the try/except, rather than all of this.

Reply:
it was the `plt.savefig`, removed all other lines from the `try`.